### PR TITLE
Player sprite

### PR DESCRIPTION
**Pull request by aerth**
With these changes, Player now idles facing down, walking is animated in 4 directions  

(`LEFT, DOWNLEFT, and UPLEFT` direction is `LEFT` animation)
(`RIGHT, DOWNRIGHT, and UPRIGHT` direction is `RIGHT` animation)

Added initial subset of shared.Action ( `A_IDLE, A_WALK, A_SLASH`, etc)

This also changes 2 things about the way mouse movement works. 

1. IF mouse is used for movement, return before checking ASDW keyboard input
2. IF chat mode, still allow mouse movement (but keep ASDW keyboard disabled)

Still using same sprite for all players, so all the players seem to be facing the same direction. This can be fixed with (among other ways)  a different version of ```msg.Update``` (serverside), or by calculating which player has moved (clientside)

### CODE REVIEW

File: client/main.go [hunk 1]
```diff
@@ -179,16 +173,20 @@ func run(protocol, addr, id string) error {
...
 		elapsed += dt
 		frameChange := 1.0 / animationRate
-		frame := int(elapsed/frameChange) % len(playerFrames)
-		playerSprite = pixel.NewSprite(playerSheet, playerFrames[frame])
+		frame := 0
+		if len(playerSprite.Frames[g.facing][g.action]) > 0 {
+			frame = int(elapsed/frameChange) % len(playerSprite.Frames[g.facing][g.action])
+		}
+		playerSprite.Sprite.Set(playerSprite.Picture, playerSprite.Frames[g.facing][g.action][frame])
+
```
Comment (ilackarms) [inline client/main.go]: i'd like us to make our own `Sprite` or `Animation` class that wraps this functionaltiy (so we can just call `Player.DrawFrame()`

in fact i think we should define an interface
```go
type Animation interface{
    Animate(deltaTime)
}
```

where `Animate()` encapsulates this behavior.

we can worry about that in a later PR, though (when we have more things to animate).

up to you if you want to define and call that interface in this PR

File: client/sprites.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+type Sprite struct {
+	Picture pixel.Picture
+	Frames  map[shared.Direction]map[shared.Action][]pixel.Rect
+	Sprite  *pixel.Sprite
+	Frame   int // current frame
+}
+
+type Atlas func() map[shared.Direction]map[shared.Action][]pixel.Rect
+
+var AtlasDefault = atlasDefault
```
Comment (ilackarms) [inline client/sprites.go]: i believe this needs to be var `AtlasDefault = atlasDefault()`

**Review by ilackarms — COMMENTED** [review]
looks great! see my comments but overall looks really good. i will give it a quick test before i approve.

**Comment by aerth** [comment]
Moved to #24

**Pull request closed without merging**